request-many/src/main/java/io/synadia/rm/RequestMany.java

```diff
@@ -0,0 +1,153 @@
+import java.util.function.Supplier;
+
+import static io.nats.client.support.NatsConstants.EMPTY_BODY;
+
+/**
+ * The RequestMany is
+ */
+public class RequestMany {
+    public static final long DEFAULT_TOTAL_WAIT_TIME_MS = 1000;
+    private static final int MIN_INTERVAL = 10;
+
+    private final Connection conn;
+    private final Supplier<String> inboxSupplier;
+    private final long totalWaitTime;
+    private final long maxStall;
+    private final long maxResponses;
+
+    public RequestMany(Builder b) {
+        this.conn = b.conn;
+        this.inboxSupplier = b.inboxSupplier == null ? conn::createInbox : b.inboxSupplier;
+        this.totalWaitTime = b.totalWaitTime;
+        this.maxStall = b.maxStall;
+        this.maxResponses = b.maxResponses;
+    }
+
+    public static Builder builder(Connection conn) {
+        return new Builder(conn);
+    }
+
+    public static class Builder {
+        private final Connection conn;
+        private Supplier<String> inboxSupplier;
+        private long totalWaitTime = DEFAULT_TOTAL_WAIT_TIME_MS;
+        private long maxStall = Long.MAX_VALUE;
+        private long maxResponses = Long.MAX_VALUE;
+
+        public Builder(Connection conn) {
+            this.conn = conn;
+        }
+
+        public Builder inboxSupplier(Supplier<String> inboxSupplier) {
+            this.inboxSupplier = inboxSupplier;
+            return this;
+        }
+
+        public Builder totalWaitTime(long totalWaitTime) {
+            this.totalWaitTime = totalWaitTime;
+            return this;
+        }
+
+        public Builder maxStall(long maxStall) {
+            this.maxStall = maxStall;
+            return this;
+        }
+
+        public Builder maxResponses(long maxResponses) {
+            this.maxResponses = maxResponses;
+            return this;
+        }
+
+        public RequestMany build() {
+            return new RequestMany(this);
+        }
+    }
+
+    public List<Message> fetch(String subject, byte[] payload) {
+        return fetch(subject, null, payload);
+    }
+
+    public List<Message> fetch(String subject, Headers headers, byte[] payload) {
+        List<Message> results = new ArrayList<>();
+        consume(subject, headers, payload, msg -> {
+            if (msg != EOD) {
+                results.add(msg);
+            }
+            return true;
+        });
+        return results;
+    }
+
+    @SuppressWarnings("DataFlowIssue")
+    public static final Message EOD = new NatsMessage("EOD", null, EMPTY_BODY);
+
+    public LinkedBlockingQueue<Message> iterate(String subject, byte[] payload) {
+        return iterate(subject, null, payload);
+    }
+
+    public LinkedBlockingQueue<Message> iterate(String subject, Headers headers, byte[] payload) {
+        final LinkedBlockingQueue<Message> q = new LinkedBlockingQueue<>();
+        conn.getOptions().getExecutor().submit(() -> {
+            consume(subject, headers, payload, msg -> {
+                if (msg != EOD) {
```

Comment:
I believe the `EOD` should also be added into the `q` as otherwise the user would not know when to stop iterating?